tests/validators/test_float.py

```diff
@@ -174,3 +178,106 @@ def test_float_key(py_and_json: PyAndJson):
     assert v.validate_test({'1.5': 1, '2.4': 2}) == {1.5: 1, 2.4: 2}
     with pytest.raises(ValidationError, match='Input should be a valid number'):
         v.validate_test({'1.5': 1, '2.5': 2}, strict=True)
+
+
+@pytest.mark.parametrize(
+    'input_value,only_finite,expected',
+    [
+        ('NaN', False, lambda x: math.isnan(x)),
```

Comment:
Smart. You could use [`FunctionCheck`](https://dirty-equals.helpmanual.io/types/other/#dirty_equals.FunctionCheck) In which case this would become

```py
FunctionCheck(math.isnan)
```

And and the below finite values could just be `0.7` or whatever.

But completely up to you, this works well too.

Reply:
Still not used to it but very convenient 👍